### DISCUSSION TITLE
List Storing in  Django Sqlite

### DISCUSSION THREAD

**Discussion by pred1824** [community/community#123528] [Programming Help]
### Body

How can I store a list in Django models?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ParasY1724** [comment]
To store a list in Django models, you can use JSONField along with the `json` module's `dump` and `load` functions for serialization and deserialization. Here's a simple example:

1. **Import JSONField:**
   First, import `JSONField` from `django.db.models`.

   ```python
   from django.db import models
   ```

2. **Define Your Model:**
   In your model class, use `JSONField` to store the list.

   ```python
   class MyModel(models.Model):
       my_list = models.JSONField(default=list)
   ```

3. **Serialize List to JSON:**
   Before saving the list to the database, you need to serialize it to JSON using `json.dumps()`.

   ```python
   import json

   my_list = [1, 2, 3, 4, 5]
   serialized_list = json.dumps(my_list)
   ```

4. **Save to Database:**
   Now, you can save the serialized list to your Django model instance.

   ```python
   my_model_instance = MyModel.objects.create(my_list=serialized_list)
   ```

5. **Deserialize JSON to List:**
   When retrieving the list from the database, you need to deserialize it back to a Python list using `json.loads()`.

   ```python
   my_model_instance = MyModel.objects.get(pk=1)
   deserialized_list = json.loads(my_model_instance.my_list)
   ```

By using `JSONField` and `json.dumps()`/`json.loads()`, you can effectively store and retrieve lists in your Django models while maintaining their structure and integrity. This approach allows you to store complex data types like lists in your database without needing to create additional models or tables.

**Reply by pred1824** [reply]
How can I display and edit a list of items associated with a Django model instance in a Django admin interface?

**Reply by ParasY1724** [reply]
To display and edit a list of items associated with a Django model instance in the Django admin interface, you can use inline formsets. Here's how you can achieve this:

1. **Define Inline Formsets:**
   Create an inline formset for the related model (in this case, the list items) within the admin definition of your main model.

   ```python
   from django.contrib import admin
   from django.forms import inlineformset_factory
   from .models import MyModel, ListItem

   class ListItemInline(admin.TabularInline):
       model = ListItem
       extra = 0

   class MyModelAdmin(admin.ModelAdmin):
       inlines = (ListItemInline,)

   admin.site.register(MyModel, MyModelAdmin)
   ```

2. **Register Models with Admin:**
   Register your models (both the main model and the related model) with the Django admin interface.

3. **Accessing in Django Admin:**
   Now, when you access the Django admin interface for your `MyModel`, you'll see an inline formset for adding, editing, and deleting list items associated with each instance of `MyModel`.

4. **Save List Items:**
   Within the Django admin interface, you can add, edit, or delete list items directly from the inline formset.

Inline formsets provide a convenient way to manage related model instances within the Django admin interface. They allow you to display and interact with related objects in a structured manner, making it easy to manage lists or other related data associated with your main model instances.
